cloudbuild.yaml

```diff
@@ -57,25 +46,65 @@ steps:
   waitFor: ['-']
 # END OF PROD BUILDING STEPS
 - name: 'gcr.io/cloud-builders/docker'
-  id: multi_arch_debian_slim
-  args: ['buildx', 'build', '--build-arg', 'CLOUD_SDK_VERSION=$_CLI_VERSION', '--platform', 'linux/arm64,linux/amd64', '-t', 'us-docker.pkg.dev/google.com/cloudsdktool/gcr.io/google-cloud-cli:slim', '-t', 'us-docker.pkg.dev/google.com/cloudsdktool/gcr.io/google-cloud-cli:$TAG_NAME-slim', '-t', 'us-docker.pkg.dev/google.com/cloudsdktool/gcr.io/google-cloud-cli:$TAG_NAME-slim-$_DATE', '-t', 'us-docker.pkg.dev/google.com/cloudsdktool/us.gcr.io/google-cloud-cli:slim', '-t', 'us-docker.pkg.dev/google.com/cloudsdktool/us.gcr.io/google-cloud-cli:$TAG_NAME-slim', '-t', 'us-docker.pkg.dev/google.com/cloudsdktool/us.gcr.io/google-cloud-cli:$TAG_NAME-slim-$_DATE', '-t', 'europe-docker.pkg.dev/google.com/cloudsdktool/eu.gcr.io/google-cloud-cli:slim', '-t', 'europe-docker.pkg.dev/google.com/cloudsdktool/eu.gcr.io/google-cloud-cli:$TAG_NAME-slim', '-t', 'europe-docker.pkg.dev/google.com/cloudsdktool/eu.gcr.io/google-cloud-cli:$TAG_NAME-slim-$_DATE', '-t', 'asia-docker.pkg.dev/google.com/cloudsdktool/asia.gcr.io/google-cloud-cli:slim', '-t', 'asia-docker.pkg.dev/google.com/cloudsdktool/asia.gcr.io/google-cloud-cli:$TAG_NAME-slim', '-t', 'asia-docker.pkg.dev/google.com/cloudsdktool/asia.gcr.io/google-cloud-cli:$TAG_NAME-slim-$_DATE', 'debian_slim/', '--push']
-  waitFor: ['multi_arch_step3']
+  id: multi_arch_debian_slim_arm
```

Comment:
This is a fairly big refactoring, what motivated us to split up the tags into separate build steps rather than building the multi-arch images at once? I agree this makes it easier to discern but at least for this review the density makes it hard for me to be 100% certain there's no breaking changes in here.

Reply:
Understood. Well the fact that it is not changing the `Dockerfile`(s), i believe the images itself are not changed at all. 

With my research I did not find any way of automatically tagging the images postfacto once the images are created. So our current approach of building both multi-arch images together will create at least two untagged images (one for `arm` and one for `amd`).

To resolve that what we are doing here is building the arm image and amd image separately (with their own tag) and then combining them into our final image with the appropriate tags.

Reply:
Why does it produce untagged images? Shouldn't it only build image tags specified by the -t flag?